prereise/gather/griddata/hifld/data_process/demand.py

```diff
@@ -0,0 +1,81 @@
+import inspect
+import os
+
+import pandas as pd
+
+from prereise.gather.griddata import hifld
+from prereise.gather.griddata.hifld import const
+
+
+def assign_demand_to_buses(substations, branch, plant, bus):
+    """Using data on population by county and ZIP code, assign demand to substations,
+    then to the lowest-voltage bus within each substation.
+    This information is added inplace as a 'Pd' column to the ``bus`` data frame.
+
+    :param pandas.DataFrame substations: table of substation data.
+    :param pandas.DataFrame branch: table of branch data.
+    :param pandas.DataFrame plant: table of plant data.
+    :param pandas.DataFrame bus: table of bus data.
+    """
+    # Load data
+    data_dir = os.path.join(os.path.dirname(inspect.getfile(hifld)), "data")
+    zip_path = os.path.join(data_dir, "uszips.csv")
+    zip_data = pd.read_csv(zip_path, dtype={"zip": "string"}).set_index("zip")
+    county_data = pd.read_csv(os.path.join(data_dir, "uscounties.csv"))
+
+    # Determine each substation's transmission capacity, then sort for selection
+    filtered_branch = branch.query("SUB_1_ID != SUB_2_ID")
+    from_cap = filtered_branch.groupby("SUB_1_ID").sum()["rateA"]
+    to_cap = filtered_branch.groupby("SUB_2_ID").sum()["rateA"]
+    sub_cap = from_cap.combine(to_cap, lambda x, y: x + y, fill_value=0)
```

Comment:
I was thinking about refactoring `calculate_substation_capacity` function in `powersimdata/design/transmission/substations.py`, which currently takes a grid object as input, but failed to come up with a compatible idea given the column names are different as well...Let's keep these lines here then.